core/src/scene/sceneLoader.cpp

```diff
@@ -520,36 +520,36 @@ bool SceneLoader::extractTexFiltering(Node& filtering, TextureFiltering& filter)
 }
 
 void SceneLoader::updateSpriteNodes(const std::string& texName,
-        std::shared_ptr<Texture>& texture, Scene& scene) {
-    auto& spriteAtlases = scene.spriteAtlases();
+        std::shared_ptr<Texture>& texture, const std::shared_ptr<Scene>& scene) {
+    auto& spriteAtlases = scene->spriteAtlases();
     if (spriteAtlases.find(texName) != spriteAtlases.end()) {
         auto& spriteAtlas = spriteAtlases[texName];
         spriteAtlas->updateSpriteNodes(texture);
     }
 }
 
 std::shared_ptr<Texture> SceneLoader::fetchTexture(const std::string& name, const std::string& url,
-        const TextureOptions& options, bool generateMipmaps, Scene& scene) {
+        const TextureOptions& options, bool generateMipmaps, const std::shared_ptr<Scene>& scene) {
 
     std::shared_ptr<Texture> texture;
 
     std::regex r("^(http|https):/");
     std::smatch match;
     // TODO: generalize using URI handlers
     if (std::regex_search(url, match, r)) {
-        scene.m_resourceLoad++;
-        startUrlRequest(url, [=, &scene](std::vector<char>&& rawData) {
+        scene->m_resourceLoad++;
+        startUrlRequest(url, [=](std::vector<char>&& rawData) {
                 auto ptr = (unsigned char*)(rawData.data());
```

Comment:
Ah - I see the issue. Shouldn't scene-loading block and wait until all resources are available?